### ISSUE TITLE
How can I provide custom templates from a package?

### ISSUE THREAD

**Issue by jason-huling** [jupyterhub/jupyterhub#2909]
I'm working on a custom spawner that is being packaged up as a python package. The spawner has a form, and up to this point I have been utilizing `options_form` to provide the form content.

At this point I'm focusing on the design of the form and have needed to add some additional javascript, css, etc. Being able to extend the `spawner.html` template seems like the cleaner approach now, vs providing these in `options_form`. 

I have read the documentation on [extending templates](https://jupyterhub.readthedocs.io/en/stable/reference/templates.html#extending-templates) which says to set `c.JupyterHub.template_paths`, but I'm not sure how to go about this correctly from my package. I am already using a jinja2 template for my `options_form` approach, so I don't have any issue making sure my template is packaged up correctly, or specifying the correct path of the template, I'm just not sure how to tell JupyterHub what the template_path is from my package.

**Comment by rcthomas** [comment]
It may or may not be possible to do this from your package but as a user I would prefer to see a minimal example `jupyterhub_config.py` included in your package or at least its documentation that outlines any of the relevant settings for your spawner to work correctly.  That would include how to set `c.JupyterHub.spawner_class` and `c.JupyterHub.template_paths` and anything else.  So it may be possible to make your package do it but documenting it rather than taking care of it may just be better anyway.

**Comment by jason-huling** [comment]
Hey @rcthomas, I completely agree with you and wouldn't have any issue with the user needing to set `template_paths` directly, but I'm still having trouble finding a clean way to do this...

I'm packaging up the templates in `<package_name>/resources/templates`, so as a workaround while I am developing the spawner I am setting the following:
```
c.JupyterHub.template_paths = ['/usr/local/lib/python3.6/dist-packages/<package_name>/resources/templates']
```
Which feels soooo unbelievably hacky :). I'm sure there is a way for me to make this cleaner and more portable, just haven't looked into it yet.

With the above said, I feel like needing setting both `spawner_class` and `template_paths` is kind of redundant. Especially if the `spawner_class` is dependent upon a specific template. I believe the user could still specify their own template either extending the base jupyterhub one and completely circumventing mine, or by extending mine.

It would be nice if jupyterhub expected templates to be in a certain location in a package and could either autoload them, or provide some way for the user to give "approval" to utilize them. Thinking something like how Django handles app templates. I didn't know if that functionality already existed and I just couldn't find it in the docs.

**Comment by jason-huling** [comment]
Did some minor digging and came up with a couple ways to make setting `template_paths` a little cleaner....

For one, I moved the template from `myspawner.resources.templates` into `myspawner.templates.jupyterhub` to specifically house templates overriding JupyterHub ones, since I have some other Jinja templates that are used by the spawner directly at the `myspanwer.templates` level.

In `jupyterhub_config.py`:
```
import os
import myspawner.templates.jupyterhub

c.JupyterHub.template_paths = [os.path.dirname(myspawner.templates.jupyterhub.__file__)]
```
Which is already much better than what I originally had... but felt it was a little verbose, so I added the following to `myspawner.templates.jupyterhub.__init__.py`:
```
import os
TEMPLATE_DIR = os.path.dirname(__file__)
```
Which lets the `jupyterhub_config.py` change to the following:
```
from myspawner.templates.jupyterhub import TEMPLATE_DIR

c.JupyterHub.template_paths = [TEMPLATE_DIR]
```
Which feels a lot more user friendly, and I'm pretty happy with.

I'll close this out, but if there are any suggestions to improve on this I would be appreciative of the feedback.
